<a href="https://colab.research.google.com/github/ish66726-a11y/Quant-Finance-Fundamentals/blob/main/notebooks/07%E3%83%96%E3%83%A9%E3%83%83%E3%82%AF%E3%82%B7%E3%83%A7%E3%83%BC%E3%83%AB%E3%82%BA%E3%83%A2%E3%83%87%E3%83%AB.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#7 Black–Scholesモデル

# 7-0 リスク中立の考え方

ブラック–ショールズ理論に入る前に、
価格評価の基本的な考え方である「リスク中立」の概念を整理する。

---

## 1.無裁定の考え方

金融市場では、**裁定（arbitrage）が存在しない**と仮定する。

裁定とは、

- 初期投資がゼロ
- 将来の損失がない
- 正の確実な利益が得られる

ような取引戦略のことである。

このような機会が存在すれば、投資家は無限に利益を得ることができてしまう。
つまり、理論モデルでは **確実に儲かる戦略は存在しない**という無裁定条件を前提とする。

---

## 2.割引価格の考え方

無リスク資産は

$$
B_t = e^{rt}
$$

と成長するとする。ここで $r$ は無リスク金利である。

任意の資産価格 $S_t$ を無リスク資産で割った

$$
\tilde{S}_t = e^{-rt} S_t
$$

を**割引価格過程**という。これは「現在価値に直した価格」と考えることができる。

---

## 3.リスク中立世界の直感的定義

無裁定市場では、割引価格は「将来に向かって平均的に増えも減りもしない」
という性質を持つ。つまり、

$$
\mathbb{E}[\tilde{S}_T \mid \mathcal{F}_t] = \tilde{S}_t
$$

が成立するような確率のもとで価格を評価できる。

このとき用いる確率を**リスク中立確率**と呼ぶ。

重要なのは、

- これは「現実の確率」ではない
- 価格評価のために用いる仮想的な確率である

という点である。

---

## 4.なぜドリフトが $r$ になるのか（直感的説明）

現実世界では、株価は

$$
dS_t = \mu S_t dt + \sigma S_t dW_t
$$

に従うと仮定する。ここで $\mu$ は実際の期待収益率である。

しかし、リスク中立確率のもとでは

$$
dS_t = r S_t dt + \sigma S_t dW_t
$$

となる。つまり、**ドリフトが $\mu$ から $r$ に置き換わる。**

その直感的な理由は次の通りである：

- デリバティブ価格は無裁定条件のもとで決まる
- 投資家のリスク選好は価格評価には直接影響しない
- 価格は「無リスクで成長する世界」で評価できる

したがって、期待収益率は無リスク金利 $r$ に一致する世界で
価格を計算することができる。

※厳密な証明はデルタヘッジの節で述べる

---

##5.まとめ

このリスク中立の考え方を用いると、  
将来のペイオフ $\Phi(S_T)$ を持つデリバティブの現在価値は

$$
V_0 = e^{-rT} \mathbb{E}^Q[\Phi(S_T)]
$$

と書くことができる。

ここで $\mathbb{E}^Q$ はリスク中立確率のもとでの期待値である。

本節で確認した手法は**期待値による価格評価**（マルチンゲール・アプローチ）と呼ばれる。

#7-1 Black–Scholes入門：リスク中立戦略とデルタヘッジ

## 1. Black–Scholesの目的

オプションの満期 $T$ における価値はあらかじめ決まっている。
これを **ペイオフ（payoff）** と呼ぶ。

- コールオプションのペイオフ
$$
\max(S_T - K, 0)
$$

- プットオプションのペイオフ
$$
\max(K - S_T, 0)
$$

  - $S_T$ ：満期時点の株価
  - $K$ ：権利行使価格


ペイオフとは、**将来の状態（ここでは $S_T$）に応じて確定する支払い額**を意味する。

つまり、満期時点の価値は明確に分かっている。しかし私たちが知りたいのは、$t < T$の時点における現在価値である。

👉 Black–Scholesモデルの目的は、**与えられたペイオフから、満期前の公正価格を理論的に導くこと**。


---

##2. 驚くべき発見：「リスク資産を組み合わせるとリスクが消える」

一般的な金融理論（モダンポートフォリオ理論、CAPM）は：

- 株価はランダムに動く  
- リスクは不可避  
- 分散投資で「銘柄固有のリスク」は消せるが、**市場全体のリスクは消せない**

という前提だった。

しかし、Black–Scholesモデルでは次を示した：

> リスク資産を特定の割合で組み合わせると、市場リスクすら完全に消える（＝無リスク化）

これは非常に革命的だった。

---

## 3. オプション価格が依存する要素

オプション価格 $V(S, t)$ は複数の要因に依存する。

#### 変数（状態量）
- $S$: 基礎資産（株価）
- $t$: 時間

#### パラメータ（モデル特性）
- $\mu$: 株の期待成長率  
- $\sigma$: ボラティリティ  
- $K$: 権利行使価格（ストライク）  
- $T$: 満期  
- $r$: 無リスク金利  

---

## 4. コールオプションとプットオプションの性質

#### コール（Call）
- 権利：未来に **買う** 権利
- 価値：$S$ が上がるほど上昇  
  → **基礎資産と正の相関**

#### プット（Put）
- 権利：未来に **売る** 権利
- 価値：$S$ が下がるほど上昇  
  → **基礎資産と負の相関**

この「相関」がデルタヘッジの根幹となる。

---
## 5. リスクを消すための特別なポートフォリオ

Black–Scholesが考えたのは、次のようなポートフォリオである：

$$
\Pi = V - \Delta S
$$

これは、

- **オプションをロング（買い）**
- **株式を $\Delta$ 単位ショート（売り）**

という組み合わせを意味する。

ここで重要なのは、コールオプションの場合、

- オプション価格 $V$ と株価 $S$ は**同じ方向に動く（正の相関）**

という点である。

この性質を利用し、適切な $\Delta$ を選ぶことで、
**オプション価格のランダムな変動を、
株式の反対売買によって完全に打ち消すことができる。**

このようなヘッジ戦略を**デルタヘッジ（Delta Hedging）** と呼ぶ。

なお、この $\Delta$ は、オプション価格の **株価に対する感応度（デルタ）** を表し、後に数学的に厳密に定義される。

その結果、このようにしてリスクを完全に消去できたならば、
このポートフォリオは無裁定条件のもとで
無リスク資産と同じ収益率で成長しなければならない。

すなわち、

$$
d\Pi = r \Pi \, dt
$$

が成立するはずである。

ここから、Black–Scholes方程式が導かれる。

---
## 6.まとめ

- Black–Scholesモデルの目的は「満期前のオプション価格」を決定すること  
- ランダムな価格変動を持つ株とオプションを組み合わせると、**リスクを完全に除去できる（デルタヘッジ）**  
- これは市場リスクすら消えるため、非常に革命的な発見  




## 7-2 Black–Scholesモデル：デルタヘッジによるリスク消去の仕組み

---

## 1. 前提：株価のモデル（幾何ブラウン運動）

まず、株価が**現実の確率のもとで**次の幾何ブラウン運動に従うと仮定する：

$$
dS = \mu S\, dt + \sigma S\, dW
$$

- $\mu S dt$：決定論的な平均成長（期待収益率）
- $\sigma S dW$：ランダム変動
- $dW$：Wiener 過程の微小変化

ここで $\mu$ は現実世界における期待成長率である。

---

## 2. 特別なポートフォリオの構築

Black–Scholesモデルの核心は次の組み合わせである：

$$
\Pi = V - \Delta S
$$

- $V(S,t)$：オプション価格
- $S$：基礎資産（株価）
- $\Delta$：ショートする株数

**オプションをロングし、株を $\Delta$ だけショートする。**

目標は、

> **このポートフォリオの確率的成分（リスク）を完全に消すこと**

である。

---

## 3. ポートフォリオの変化：$d\Pi$

ポートフォリオの微小変化は

$$
d\Pi = dV - \Delta\, dS
$$

で与えられる。

ここで、$V$ も $S$ も確率過程であるため、
$dV$ を求めるには伊藤の補題を用いる必要がある。

---

## 4. 伊藤の補題によるオプション価格の変化

オプション価格 $V(S,t)$ は株価 $S$ と時間 $t$ の関数である。
株価が確率過程である以上、$V$ も確率過程となる。

伊藤の補題（二変数版）より、

$$
dV
=
V_t\,dt
+
V_S\,dS
+
\frac{1}{2}V_{SS}(dS)^2
$$

が成り立つ。

株価は

$$
dS = \mu S\,dt + \sigma S\,dW
$$

に従う。

微小項の性質より、

- $(dt)^2$ および $dt\,dW$ は無視できる
- ただし $(dW)^2 = dt$ は無視できない

したがって、

$$
(dS)^2 \approx \sigma^2 S^2\,dt
$$

となる。

これを代入すると、

$$
dV
=
\left(
V_t + \mu S V_S + \frac{1}{2}\sigma^2 S^2 V_{SS}
\right)dt
+
\sigma S V_S\,dW
$$

を得る。

右端の $dW$ を含む項が、
オプション価格の**確率的なリスク成分**である。

---

## 5. ポートフォリオの変化

ポートフォリオ

$$
\Pi = V - \Delta S
$$

の微小変化は

$$
d\Pi = dV - \Delta\,dS
$$

である。

これに

$$
dS = \mu S\,dt + \sigma S\,dW
$$

を代入すると、

$$
\begin{aligned}
d\Pi
&=
\left(
V_t + \mu S V_S + \frac{1}{2}\sigma^2 S^2 V_{SS}
- \Delta \mu S
\right)dt
\\
&\quad+
\left(
\sigma S V_S - \Delta \sigma S
\right)dW
\end{aligned}
$$

となる。

---

## 6. 確率項の消去：デルタヘッジ

ここで

$$
\Delta = V_S
$$

と選ぶと、

$$
\sigma S V_S - \Delta \sigma S = 0
$$

となり、$dW$ の項（確率的成分）が完全に消える。

これが **デルタヘッジ（Delta Hedging）** である。

重要なのは、

> 2つのリスク資産を適切な割合で組み合わせることで、
> その瞬間の確率的変動を完全に打ち消すことができる

という点である。

---

## 7. 無裁定条件と無リスク成長

確率項が消えたならば、
ポートフォリオ $\Pi$ は瞬間的に無リスクとなる。

無裁定条件のもとでは、
無リスク資産は無リスク金利 $r$ で成長しなければならない。

したがって、

$$
d\Pi = r \Pi\, dt
$$

が成立する。

この関係式から、Black–Scholes方程式が導かれる。

---

## 8. デルタは動的に変化する

$$
\Delta = V_S
$$

は時間とともに変化する。

- $S$ が変わる
- $t$ が進む
- 市場条件も変化する

したがって、

> ヘッジは常に再調整されなければならない  
> これを**動的デルタヘッジ**という。

# 7-3 Black–Scholes方程式の導出  


## 1. リスクを消した結果：ポートフォリオは瞬間的に確定的になる

デルタヘッジ後のポートフォリオ：

$$
\Pi = V - \Delta S
$$

デルタ $\Delta = V_S$ を選ぶと、

$$
d\Pi \text{ から拡散項 } dW \text{ が完全に消える}
$$

したがって、このポートフォリオは

> **瞬間的に確率的変動を持たず、決定論的な変化のみを持つ。**

重要なのは、これは「将来永遠に確定的になる」という意味ではなく、  
その瞬間のランダム成分が消去されるという意味である。

---

## 2. ノーアービトラージ原理（No-Arbitrage Principle）

金融市場の基本原則は無裁定条件である。

無裁定条件のもとでは、

> 無リスク資産と同じリスク特性を持つポートフォリオは、  
> 無リスク金利 $r$ で成長しなければならない。

もしこれが成り立たなければ、

- リスクなしで市場平均以上の収益が得られる  
- 借入による無限利益が可能になる  

これは裁定機会となり、市場はそれを許さない。

したがって、

$$
d\Pi = r \Pi\, dt
$$

が成立する。

---

## 3. ポートフォリオの決定論的変化

前章の結果より、デルタヘッジ後の $d\Pi$ の決定論的成分は：

$$
d\Pi
=
\left(
V_t
+ \frac12 \sigma^2 S^2 V_{SS}
+ \mu S V_S
- \Delta \mu S
\right) dt
$$

デルタ $\Delta = V_S$ を代入すると、

$$
d\Pi
=
\left(
V_t + \frac12 \sigma^2 S^2 V_{SS}
\right) dt
$$

ここで注目すべきは、

> 株価の期待成長率 $\mu$ が完全に消えている点である。

つまり、オプション価格は基礎資産の期待収益率 $\mu$ に依存しない。

---

## 4. ノーアービトラージ条件との結合

ポートフォリオは

$$
\Pi = V - S V_S
$$

であるから、

$$
d\Pi = r \Pi\, dt
$$

すなわち、

$$
V_t + \frac12 \sigma^2 S^2 V_{SS}
=
r (V - S V_S)
$$

が成立する。

---

## 5. Black–Scholes偏微分方程式（Black–Scholes PDE）

整理すると、

$$
V_t
+ \frac12 \sigma^2 S^2 V_{SS}
+ r S V_S
- r V
=
0
$$

これが金融工学で最も重要な方程式の一つ、**Black–Scholes方程式**である。

---

## 6. なぜこの方程式は重要なのか？

- この PDE の解がヨーロピアン・オプション価格を与える  
- 無裁定条件のみから導かれた価格決定式である  
- 期待成長率 $\mu$ に依存しない  
- 満期でのペイオフを境界条件として与えることで  
  → Black–Scholesの解析解が得られる  

さらに重要なのは、

> この結果は、前節で述べたリスク中立評価と
> 本質的に同じ価格を別の方法（デルタヘッジ）から導いている
> ことを意味する。

---

## 7. まとめ

- デルタヘッジにより確率項（$dW$）が消える  
- ポートフォリオは瞬間的に無リスクとなる  
- 無裁定条件より成長率は $r$  
- その条件が Black–Scholes PDE を強制する  
- PDE の解がオプション価格そのものである  

#7-4 Black–Scholes方程式の解析解

## 1. Black–Scholes方程式

オプション価格 $V(S,t)$ は次の PDE を満たす：

$$
V_t + \frac12 \sigma^2 S^2 V_{SS} + r S V_S - r V = 0
$$

これは数学的には **放物型 PDE**。

> 代表例：熱方程式（heat equation）、拡散方程式（diffusion equation）

金融数学においても同じ数学構造が出てくる。

---
## 2. PDE を直接解く必要は「ない」

Black–Scholes方程式の解法は高度な数学を含む：

- 伊藤積分
- 変数変換  
- 熱方程式への写像
- ガウス積分

しかし、**解析解**（closed-form solution）は非常にシンプルで、これをそのまま使えばよい。

---
## 3. 標準正規分布の累積分布関数 $N(x)$ の定義

ブラック＝ショールズ方程式の解析解では **標準正規分布の累積分布関数（CDF）** を使う：

$$
N(x) = \frac{1}{\sqrt{2\pi}}
\int_{-\infty}^{x} e^{-y^2 / 2} \, dy
$$

実装では NumPy や SciPy に含まれているので、自分で積分する必要はない。

---
## 4. 必要となるパラメータ $d_1$ と $d_2$

Black–Scholes方程式の解析解では、まず次の二つを計算する：

#### $d_1$ の定義

$$
d_1 =
\frac{
\ln\left(\frac{S}{K}\right)
+ \left(
r + \frac12 \sigma^2
\right)(T - t)
}{
\sigma \sqrt{T - t}
}
$$

#### $d_2$ の定義

$$
d_2
=
\frac{
\ln\left(\frac{S}{K}\right)
+ \left(
r - \frac12 \sigma^2
\right)(T - t)
}{
\sigma \sqrt{T - t}
} \\
= d_1 - \sigma \sqrt{T - t}
$$

意味：

- $\ln(\frac{S}{K})$：現在の株価と行使価格の相対関係  
- $r$：無リスク金利  
- $\sigma$：ボラティリティ  
- $T-t$：満期までの時間  

---
##5. Black–Scholes方程式の解析解

コールオプション（Call）の価格：

$$
C = S N(d_1) - K e^{-r(T - t)} N(d_2)
$$


プットオプション（Put）の価格：

$$
P = K e^{-r(T - t)} N(-d_2) - S N(-d_1)
$$

---
##6. まとめ

- Black–Scholes 方程式は高度だが、最終解は単純  
- 標準正規分布の累積分布関数 $N(x)$ が重要  
- $d_1$,$d_2$ を計算し、コール/プット公式に代入するだけ  






# 【補足】解析解の導出

## 1. 出発点：Black–Scholes方程式

コールオプションに対して、価格 $V(S_t,t)$ は次の偏微分方程式を満たす。

$$
\frac{\partial V}{\partial t}
+r S_t \frac{\partial V}{\partial S_t}
+\frac12\sigma^2 S_t^2 \frac{\partial^2 V}{\partial S_t^2}
=
rV
$$

満期条件は

$$
V(S_T,T)=\max(S_T - K, 0)
$$

である。

---

## 2. 変数変換

### (1) 残存期間

$$
\tau = T - t
$$

とおくことで、時間微分の符号が反転する。

$$
\frac{\partial}{\partial t}
=
- \frac{\partial}{\partial \tau}
$$





### (2) 対数変数＋ドリフト補正

原資産価格の対数を取り、さらに一次微分項（ドリフト項）を打ち消すための補正項を加えた新しい変数 $u$ を定義する。

$$
u
=
\log\!\left(\frac{S_t}{K}\right)
+
\left(r-\frac{\sigma^2}{2}\right)\tau
$$



### (3) 割引因子の導入

オプション価格を無リスク金利で割り引いた新しい関数 $y(u,\tau)$ を定義する。

$$
V(S_t,t)
=
e^{-r\tau} y(u,\tau)
$$



---

## 3. 熱方程式への帰着

チェインルールにより

$$
V_{S_t}
=
\frac{e^{-r\tau}}{S_t} y_u
$$

$$
V_{S_tS_t}
=
\frac{e^{-r\tau}}{S_t^2}
(y_{uu}-y_u)
$$

$$
V_t
=
e^{-r\tau}
\left(
r y
-
y_\tau
-
\left(r-\frac{\sigma^2}{2}\right)y_u
\right)
$$

これらをBlack–Scholes方程式に代入すると、
一次微分項が完全に消去され、次の熱方程式を得る。

$$
y_\tau
=
\frac12\sigma^2 y_{uu}
$$

---

## 4. 初期条件

満期 $t=T$ は $\tau=0$ に対応する。

$$
y(u,0)=V(S_T,T)
$$

$\tau=0$ では

$$
S_T = K e^u
$$

であるため、

$$
y(u,0)
=
\max(K(e^u-1), 0)
$$

すなわち

$$
y(u,0)=
\begin{cases}
K(e^u-1) & u\ge0 \\
0 & u<0
\end{cases}
$$

---


## 5. 分離解とフーリエ展開

熱方程式

$$
y_\tau
=
\frac12\sigma^2 y_{uu}
$$

を解く。

解を分離形

$$
y(u,\tau)=U(u)W(\tau)
$$

と仮定する。

これを代入すると

$$
U(u)W'(\tau)
=
\frac12\sigma^2 U''(u)W(\tau)
$$

両辺を $\frac12\sigma^2 U(u)W(\tau)$ で割ると、

$$
\frac{W'(\tau)}{\frac12\sigma^2 W(\tau)}
=
\frac{U''(u)}{U(u)}
$$

左辺は $\tau$ のみ、右辺は $u$ のみの関数である。

したがって、これらはある定数に等しくなければならない。
この定数を $-\omega^2$（$\omega \ge 0$）とおく。

$$
\frac{W'(\tau)}{\frac12\sigma^2 W(\tau)}
=
\frac{U''(u)}{U(u)}
=
-\omega^2
$$

よって2つの常微分方程式を得る。



### (1) $\tau$ に関する方程式

$$
W'(\tau)
=
-\frac12\sigma^2 \omega^2 W(\tau)
$$

$$
W(\tau)
=
\exp\!\left(
-\frac12\sigma^2 \omega^2 \tau
\right)
$$

### (2) $u$ に関する方程式

$$
U''(u)
+
\omega^2 U(u)
=
0
$$

その解は

$$
U(u)
=
C(\omega)\cos(\omega u)
+
D(\omega)\sin(\omega u)
$$

である。



線形性より、一般解は $\omega$ に関する重ね合わせで表される：

$$
y(u,\tau)
=
\int_0^\infty
\left[
C(\omega)\cos(\omega u)
+
D(\omega)\sin(\omega u)
\right]
\exp\!\left(
-\frac12\sigma^2 \omega^2 \tau
\right)
\, d\omega
$$

---

## 6. 初期条件の適用とガウス核表示

ここで初期条件

$$
y(u,0)=g(u)
$$

を適用する。

フーリエ変換の理論より、任意の関数 $g(u)$ は
$\cos(\omega u)$ および $\sin(\omega u)$ の重ね合わせで表せる。

したがって、上式に初期条件を適用し、
フーリエ逆変換を行うと、
解は次の「ガウス核（熱核）」との畳み込み表示に帰着する：

$$
y(u,\tau)
=
\frac{1}{\sqrt{2\pi\sigma^2\tau}}
\int_{-\infty}^{\infty}
g(a)
\exp\!\left(
-\frac{(a-u)^2}{2\sigma^2\tau}
\right)
da
$$

これは熱方程式のグリーン関数

$$
\frac{1}{\sqrt{2\pi\sigma^2\tau}}
\exp\!\left(
-\frac{(a-u)^2}{2\sigma^2\tau}
\right)
$$

を用いた表現であり、
初期条件を時間 $\tau$ だけ拡散させた形になっている。


ここでコールの初期条件

$$
g(a)=\max(K(e^a-1),0)
$$

を代入する。

$ g(a)=0 \quad (a<0) $ であるため、

$$
y(u,\tau)
=
\frac{1}{\sqrt{2\pi\sigma^2\tau}}
\int_{0}^{\infty}
K(e^a-1)
\exp\!\left(
-\frac{(a-u)^2}{2\sigma^2\tau}
\right)
da
$$


### (1) A と B に分解

$$
y=A-B
$$

$$
A=
\frac{K}{\sqrt{2\pi\sigma^2\tau}}
\int_0^\infty
e^a
\exp\!\left(
-\frac{(a-u)^2}{2\sigma^2\tau}
\right)
da
$$

$$
B=
\frac{K}{\sqrt{2\pi\sigma^2\tau}}
\int_0^\infty
\exp\!\left(
-\frac{(a-u)^2}{2\sigma^2\tau}
\right)
da
$$

---

## 7. 元の変数への復元

$$
V(S_t,t)=e^{-r\tau}y(u,\tau)
$$

を用いて整理すると、

$$
V(S_t,t)
=
S_t N(d_1)
-
K e^{-r\tau} N(d_2)
$$

ただし

$$
d_2
=
\frac{
\log(S_t/K)
+
\left(r-\frac12\sigma^2\right)\tau
}
{\sigma\sqrt{\tau}}
$$

$$
d_1
=
d_2
+
\sigma\sqrt{\tau}
=
\frac{
\log(S_t/K)
+
\left(r+\frac12\sigma^2\right)\tau
}
{\sigma\sqrt{\tau}}
$$

である。
以上により、Black–Scholes方程式の解析解が導出された。


---

## 8.プット・コール・パリティ（プットオプションの場合）

これまでにコールオプションの解析解

$$
\boxed{
C = S_t N(d_1) - K e^{-r\tau} N(d_2)
}
$$

を導出した。

ここではプットオプションの価格を導出する。


### a. プットの満期条件

プットオプションの満期ペイオフは

$$
V(S_T,T) = \max(K - S_T, 0)
$$

である。

Black–Scholes方程式自体はコールと同一であり、

$$
V_t + r S_t V_{S_t}
+ \frac12\sigma^2 S_t^2 V_{S_tS_t}
=
rV
$$

を満たす。

変わるのは終端条件のみである。



### b. プット・コール・パリティの導出

次の2つのポートフォリオを考える。

### ポートフォリオA
- コール1枚

満期価値：

$$
\max(S_T - K, 0)
$$

### ポートフォリオB
- プット1枚
- 株式1株

満期価値：

$$
\max(K - S_T, 0) + S_T
$$



### 満期での比較

#### ケース1：$S_T > K$

$$
A = S_T - K
$$

$$
B = S_T
$$

よって

$$
B - A = K
$$



#### ケース2：$S_T \le K$

$$
A = 0
$$

$$
B = (K - S_T) + S_T = K
$$

よって

$$
B - A = K
$$

### 結論

常に

$$
B = A + K
$$

が成立する。

したがって現在価値に割り引くと

$$
S_t + P = C + K e^{-r\tau}
$$

整理すると

$$
\boxed{
C - P = S_t - K e^{-r\tau}
}
$$

これを **プット・コール・パリティ** という。



### c. プット価格の導出

パリティを用いれば、

$$
P = C - S_t + K e^{-r\tau}
$$

である。

コール価格

$$
C = S_t N(d_1) - K e^{-r\tau} N(d_2)
$$

を代入すると

$$
P
=
K e^{-r\tau} N(-d_2)
-
S_t N(-d_1)
$$

を得る。


### d. まとめ

プットオプションの解析解は

$$
\boxed{
P(S_t,t)
=
K e^{-r\tau} N(-d_2)
-
S_t N(-d_1)
}
$$

である。

ここで

$$
d_1
=
\frac{
\log(S_t/K)
+
\left(r+\frac12\sigma^2\right)\tau
}
{\sigma\sqrt{\tau}}
$$

$$
d_2 = d_1 - \sigma\sqrt{\tau}
$$

である。




#7-5 オプション価格の感応度指標 ― Greeks（グリークス）

Greeks は、オプション価格が何に対してどれだけ敏感かを測る尺度であり、  
ヘッジング・リスク管理・ポートフォリオ最適化で中心的な役割を果たす。

---

## 1. Delta $\Delta$（デルタ）

#### 定義  
デルタはオプション価格 $V$ の株価 $S$ に対する一次微分：

$$
\Delta = \frac{\partial V}{\partial S}
$$

意味：  
- **オプションの価格が、株価のわずかな変化に対してどれくらい変わるか**
- オプションの「株的感応度」

#### 重要な性質  
Black–Scholesのデルタヘッジで登場したように：

> $\Delta$ を株のショート量として用いると、リスクが完全に消える（$dW$ が消える）

すなわち：

- ポートフォリオ $\Pi = V - \Delta S$  の確率項を 0 にできる  
- デルタヘッジは **リスクの完全除去** を目的とした戦略

---

## 2. Gamma $\Gamma$（ガンマ）

#### 定義  
ガンマはオプション価格の二次微分：

$$
\Gamma = \frac{\partial^2 V}{\partial S^2}
$$

意味：

- **デルタが株価の変化に対してどれだけ変わるか（デルタの感応度）**
- デルタの変化率（カーブの強さ）

#### ガンマの役割  
- ガンマが大きいほど、デルタ値が急激に変動する  
- つまり **ヘッジの調整頻度が増える**
- デルタヘッジが「動的」でなければいけない理由

---

## 3. Theta $\Theta$（シータ）

#### 定義  
オプション価格の時間に対する微分：

$$
\Theta = \frac{\partial V}{\partial t}
$$

意味：

- **時間の経過によって、オプション価格がどれだけ減少するか**
- 特にオプションは時間価値を持つため、多くの場合 $\Theta < 0$


---

## 4. Vega（ベガ）

#### 定義  
ボラティリティ $\sigma$ に対する感応度：

$$
\text{Vega} = \frac{\partial V}{\partial \sigma}
$$

意味：

- **ボラティリティが変わるとオプション価格がどれだけ動くか**

ポイント：

- ボラティリティは市場予測で大きくブレる  
- そのため **ボラティリティリスクは非常に重要**

---

## 5. Rho $\rho$（ロー）（補足）

金利 $r$ に対する感応度：

$$
\rho = \frac{\partial V}{\partial r}
$$

金利変化がオプション価格に与える影響を示す。

---

## 6. Black–Scholes方程式 を Greeks で書き換える

Black–Scholes 方程式：

$$
V_t + \frac12\sigma^2 S^2 V_{SS} + r S V_S - rV = 0
$$

Greeks を用いると：

- $V_t = \Theta$  
- $V_S = \Delta$  
- $V_{SS} = \Gamma$

これにより、

$$
\Theta + \frac12 \sigma^2 S^2 \Gamma + r S \Delta - rV = 0
$$

このように、PDE を「Greeks の組み合わせ」として表現できる。

---

## 7. Greeks のまとめ

| Greek | 数式 | 意味 |
|-------|------|------|
| $\Delta$ | $\frac{\partial V }{ \partial S}$ | 株価への感応度（一次） |
| $\Gamma$ | $\frac{\partial^2 V}{\partial S^2}$ | デルタの感応度（二次） |
| $\Theta$ | $\frac{\partial V}{\partial t}$ | 時間経過の影響（タイムディケイ） |
| Vega | $\frac{\partial V}{\partial \sigma}$ | ボラティリティの影響 |
| $\rho$ | $\frac{\partial V}{\partial r}$ | 金利の影響 |


# 7-6 Black–Scholes モデルで儲けるには？

Black–Scholes モデルは、利益獲得にも使われるが、実務では主に**リスク管理と分析の道具**として機能している。

- Greeks を計算してポートフォリオをヘッジする  
- 理論価格と市場価格のズレを測る  
- インプライド・ボラティリティを可視化する  

つまり、

> Black–Scholes は「価格を当てる公式」ではなく  
> 「リスクと市場の認識を分解する装置」である。

---

## 立場による使い方の違い

### セルサイド

**目的：リスクを抑えながら安定収益を得る**

- Greeks によるヘッジ
- 理論価格を使って適正な提示価格を算出
- ミスプライスで大勝負は基本しない

> リスクを消す側



### バイサイド

**目的：リスクを取り、超過収益を狙う**

- 理論価格と市場価格のズレを分析
- インプライド・ボラティリティを評価
- 割安なら買い、割高なら売る

> リスクを取りに行く側

---

## LTCMの事例

Black–Scholes を用いて利益を上げる、という発想は理論上は明快である。

- 理論価格と市場価格のズレを見つける  
- 相関を利用してロングとショートを組む  
- リスクをヘッジしたうえで価格収束を待つ  

LTCMは、この考え方を大規模に実践した代表的事例である。

1994年に設立されたLTCMは、Black–Scholesモデルやペアトレーディングを用いた  市場中立型の相対価値戦略を展開した。

設立から約3年間、年率約40％という高いリターンを記録したが、1998年のロシア金融危機などにより相関構造が崩壊し、巨額損失を計上して事実上破綻した。

この事例が示すのは、

> モデルはリスクを消すのではなく、  
> リスクの「形」を変えるだけである

という点である。

Black–Scholes はミスプライスを測る道具であり、市場中立戦略はリスクを抑える設計を持つ。しかし、極端な市場環境では想定していなかったテールリスクが顕在化する。

したがって、Black–Scholesで「儲ける」とは、

- 理論価格を盲信することではなく  
- どのリスクを引き受けているのかを理解した上で  
- そのリスクに対価があると判断すること

を意味する。

---

## まとめ

| 行為 | セルサイド | バイサイド |
|------|------------|------------|
| Greeksでヘッジ | ◎（必須） | ◯（戦略次第） |
| 理論価格の算出 | ◎（価格提示のため） | ◎（分析のため） |
| ミスプライス活用 | △（限定的） | ◎（主戦場） |



#7-7 Black–Scholesモデル実装（Python）  

## 1. 必要ライブラリの読み込み

Black–Scholesモデルの公式は、正規分布の累積分布関数（CDF）を使用するため、SciPy の `stats.norm.cdf` を使う。また NumPy の `log`, `exp`, `sqrt` も利用する。

---

## 2. d1, d2 の計算式

Black–Scholesモデルでは、まず以下の 2 つの値を計算する：

 $$d_1 = \frac{\ln(\frac{S}{K}) + (r + \frac{1}{2}\sigma^2)T}{\sigma\sqrt{T}}$$

 $$d_2 = d_1 - \sigma\sqrt{T}$$

ここで  

- $S$ … 現在の株価  
- $K$ … 権利行使価格（ストライク）  
- $T$ … 残存期間（年）  
- $r$ … 無リスク金利  
- $\sigma$ … ボラティリティ  

---

## 3. Black–Scholes方程式の解析解

Black–Scholesのコール価格公式：

$$C = S\,N(d_1)\;-\;K e^{-rT} N(d_2)$$



Black–Scholesのプット価格公式：

$$P = K e^{-rT} N(-d_2)\;-\;S\,N(-d_1)$$


---
## 4. Python実装例：オプション価格の計算






In [ ]:
from scipy import stats
from numpy import log, exp, sqrt


# --------------------------------
# コールオプション価格を計算する関数
# --------------------------------
def call_option_price(S, K, T, r, sigma):
    """
    S : 現在の株価
    K : ストライク価格（行使価格）
    T : 満期までの時間（年）
    r : 無リスク金利（年率）
    sigma : ボラティリティ（年率）
    """

    # Black–Scholes の中心となるパラメータ
    d1 = (log(S / K) + (r + 0.5 * sigma**2) * T) / (sigma * sqrt(T))
    d2 = d1 - sigma * sqrt(T)

    # コールオプション価格公式
    call = S * stats.norm.cdf(d1) - K * exp(-r * T) * stats.norm.cdf(d2)

    return call


# --------------------------------
# プットオプション価格を計算する関数
# --------------------------------
def put_option_price(S, K, T, r, sigma):

    # d1, d2 はコールと同じ式
    d1 = (log(S / K) + (r + 0.5 * sigma**2) * T) / (sigma * sqrt(T))
    d2 = d1 - sigma * sqrt(T)

    # プットオプション価格公式
    put = K * exp(-r * T) * stats.norm.cdf(-d2) - S * stats.norm.cdf(-d1)

    return put


# --------------------------------
# 実行部分
# --------------------------------

# パラメータ設定
S = 100      # 現在の株価
K = 100      # 行使価格
T = 1        # 満期まで1年
r = 0.05     # 無リスク金利 5%
sigma = 0.20 # ボラティリティ 20%

# 価格計算
call = call_option_price(S, K, T, r, sigma)
put = put_option_price(S, K, T, r, sigma)

# 結果表示
print("【Black–Scholesモデルによるオプション価格の計算結果】")
print(f"現在の株価 S = {S}")
print(f"行使価格 K = {K}")
print(f"残存期間 T = {T} 年")
print(f"無リスク金利 r = {r*100}%")
print(f"ボラティリティ σ = {sigma*100}%")
print("-------------------------------------------")
print(f"コールオプション理論価格：{call:.4f}")
print(f"プットオプション理論価格：{put:.4f}")


【Black–Scholesモデルによるオプション価格の計算結果】
現在の株価 S = 100
行使価格 K = 100
残存期間 T = 1 年
無リスク金利 r = 5.0%
ボラティリティ σ = 20.0%
-------------------------------------------
コールオプション理論価格：10.4506
プットオプション理論価格：5.5735


---

## 5. 動作確認（他サイトとの一致）

オンライン計算ツールで同じ値を計算すると次と一致する：

- **Call ≈ 10.45**
- **Put  ≈ 5.57**

これにより、実装が正しく機能していることが確認できる。

---

## 6. Black–Scholesモデルが強力な理由

数学的導出は  

- 確率微分方程式 (SDE)  
- 幾何ブラウン運動  
- Ito の補題  
- PDE（偏微分方程式）

など高度な内容が含まれるが、**実際の実装はわずか数行で済む。**

だからこそ、実務の現場ではBlack–Scholesは非常に強力で使われ続けている。

---

## 7. まとめ

- $d_1$、$d_2$ の計算が最初のステップ  
- SciPy により正規分布は自前実装不要  
- Python でコール/プット価格が簡潔に実装できる  
- 複雑な数学的背景を意識せず利用可能  
- 実務的にも非常に使われる標準モデル  




# 7-8 モンテカルロ法によるオプション価格計算の準備（理論編）

---

## 1. 基本アイデア：将来の株価が必要

オプション価格は、満期におけるペイオフに依存する。

コールオプション：
$$
\max(S_T - K, 0)
$$

プットオプション：
$$
\max(K - S_T, 0)
$$

ここで $S_T$ は **満期 $T$ における株価** である。

したがって、価格を求めるためには  
将来の株価 $S_T$ の分布が必要になる。

---

## 2. リスク中立評価式

前節で導いたように、無裁定条件のもとでは  
オプション価格はリスク中立確率のもとで

$$
V_0 = e^{-rT}\,\mathbb{E}^Q[\Phi(S_T)]
$$

と書くことができる。

ここで

- $\Phi(S_T)$：満期時点のペイオフ  
- $\mathbb{E}^Q$：リスク中立確率のもとでの期待値  

である。

モンテカルロ法とは、

> この期待値を数値的に近似する方法

である。

---

## 3. 株価モデル：幾何ブラウン運動（GBM）

現実世界では株価は次のSDEに従うと仮定する：

$$
dS_t = \mu S_t\,dt + \sigma S_t\,dW_t
$$

しかし、無裁定条件のもとで価格を評価する際には、
リスク中立測度のもとでのダイナミクスを用いる。

その結果、株価は

$$
dS_t = r S_t\,dt + \sigma S_t\,dW_t
$$

に従う。

---

## 4. リスク中立測度下での株価分布

上のSDEを解くと、

$$
S_T
=
S_0 \exp\!\left[
\left( r - \frac12\sigma^2 \right)T
+
\sigma \sqrt{T} Z
\right],
\quad
Z \sim \mathcal{N}(0,1)
$$

となる。

👉 $S_T$ は対数正規分布に従う。

---

## 5. モンテカルロ法の手順

1. 標準正規乱数 $Z$ を大量に生成  
2. 上式を用いて $S_T$ を計算  
3. 各シナリオでペイオフを計算  

コールオプション：
$$
P_i = \max(S_T^{(i)} - K,\; 0)
$$

プットオプション：
$$
P_i = \max(K - S_T^{(i)},\; 0)
$$

---

## 6. 割引して現在価値へ

リスク中立評価式に従い、

$$
V_0 \approx
e^{-rT}
\cdot
\frac{1}{N}
\sum_{i=1}^{N}
\Phi(S_T^{(i)})
$$

と近似する。

これがモンテカルロ法によるオプション価格である。

---

## 7. なぜモンテカルロ法が重要か？

- ブラック–ショールズ解析解が存在しないケースに対応可能  
- ペイオフが複雑でも扱える  
- パス依存型（Asian, Barrier, Lookback など）にも適用可能  

👉 モンテカルロ法はリスク中立期待値を数値的に計算するための汎用的なツールである。

# 7-9 モンテカルロ法によるオプション価格シミュレーション（Call / Put）

---

## 1. 理論：リスク中立評価とモンテカルロ

無裁定条件のもとでは、ヨーロピアン・オプション価格はリスク中立確率のもとで

$$
V_0 = e^{-rT}\,\mathbb{E}^Q[\Phi(S_T)]
$$

と表される。ここで

- $r$：無リスク金利  
- $T$：満期  
- $\Phi(S_T)$：満期ペイオフ  
- $\mathbb{E}^Q[\cdot]$：リスク中立確率のもとでの期待値  

である。

モンテカルロ法は、この期待値を**大量サンプルの平均で近似する方法**である。

---

## 2. リスク中立測度のもとでの株価分布（GBMの解析解）

リスク中立測度のもとでは、株価は

$$
dS_t = rS_t\,dt + \sigma S_t\,dW_t
$$

に従う（ドリフトが $r$ になる）。

このSDEの解は

$$
S_T
=
S_0 \exp\left[
\left(r - \tfrac{1}{2}\sigma^2\right)T
+
\sigma\sqrt{T}\,Z
\right],
\quad
Z \sim \mathcal{N}(0,1)
$$

となる。

---

## 3. コールとプットのペイオフ

満期におけるペイオフは次で定義される。

コール：
$$
\Phi_{\text{Call}}(S_T) = \max(S_T - K,\;0)
$$

プット：
$$
\Phi_{\text{Put}}(S_T) = \max(K - S_T,\;0)
$$

両者の違いは **ペイオフ関数のみ**であり、株価モデルおよび割引評価の枠組みは共通である。

---

## 4. Python 実装例：モンテカルロ法によるオプション価格シミュレーション（Call / Put）





In [ ]:
import numpy as np


class OptionPricing:

    def __init__(self, S0, K, T, r, sigma, iterations):
        """
        S0         : 初期株価
        K          : 行使価格（ストライク）
        T          : 満期までの期間（年）
        r          : 無リスク金利（年率）
        sigma      : ボラティリティ（年率）
        iterations : シミュレーション回数
        """
        self.S0 = S0
        self.K = K
        self.T = T
        self.r = r
        self.sigma = sigma
        self.iter = iterations


    # ==================================
    # コールオプション価格（Monte Carlo）
    # ==================================
    def call_option_simulation(self):

        # --- 標準正規乱数を生成 ---
        Z = np.random.normal(0, 1, self.iter)

        # --- 満期株価 S_T ---
        # S_T = S0 * exp[(r - 0.5σ²)T + σ√T Z]
        S_T = self.S0 * np.exp(
            (self.r - 0.5 * self.sigma**2) * self.T
            + self.sigma * np.sqrt(self.T) * Z
        )

        # --- コールのペイオフ ---
        payoff = np.maximum(S_T - self.K, 0)

        # --- 現在価値へ割引 ---
        price = np.exp(-self.r * self.T) * np.mean(payoff)

        return price


    # ==================================
    # プットオプション価格（Monte Carlo）
    # ==================================
    def put_option_simulation(self):

        Z = np.random.normal(0, 1, self.iter)

        S_T = self.S0 * np.exp(
            (self.r - 0.5 * self.sigma**2) * self.T
            + self.sigma * np.sqrt(self.T) * Z
        )

        payoff = np.maximum(self.K - S_T, 0)

        price = np.exp(-self.r * self.T) * np.mean(payoff)

        return price


# ==========================================
# メイン処理
# ==========================================

model = OptionPricing(
    S0 = 100,     # 現在の株価
    K = 100,      # 行使価格
    T = 1,        # 満期まで1年
    r = 0.05,     # 無リスク金利 5%
    sigma = 0.20, # ボラティリティ 20%
    iterations=10000
)

call_price = model.call_option_simulation()
put_price = model.put_option_simulation()

print("【モンテカルロ法によるオプション価格評価】")
print(f"コールオプション価格：{call_price:.4f}")
print(f"プットオプション価格：{put_price:.4f}")


【モンテカルロ法によるオプション価格評価】
コールオプション価格：10.6476
プットオプション価格：5.4633


---    
## 5. 出力例（典型値）

- **Call ≈ 10.45**
- **Put  ≈ 5.57**

（※ シミュレーション回数を増やすと、モンテカルロ推定値は Black–Scholesの解析解に収束する。）

---

## 6. まとめ

- コールとプットの違いは **ペイオフ関数のみ**
- 株価は **リスク中立測度のもとで GBM に従う**
- オプション価格は
  $$
  V_0 = e^{-rT}\,\mathbb{E}^Q[\Phi(S_T)]
  $$
  で与えられる
- モンテカルロ法では $S_T$ を大量に生成し、
  ペイオフの平均を計算して $e^{-rT}$ で割引することで **現在のオプション価格を数値的に近似する**